### PR TITLE
DCS-1604 Hook up search and address PR comments 

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/RecentArrivalsService.kt [hunk 1]
```diff
@@ -4,33 +4,35 @@ import org.springframework.data.domain.Page
...
   fun getArrivals(
     prisonId: String,
-    fromDate: LocalDate,
-    toDate: LocalDate,
-    pageSize: Int,
-    page: Int
+    dateRange: Pair<LocalDate, LocalDate>,
+    page: PageRequest,
+    query: String? = null
   ): Page<RecentArrival> {
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/RecentArrivalsService.kt:24]: didn't want to add another parameter as list was getting long so moved to use a pair and page object

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/search/FuzzyScorer.kt [hunk 2]
```diff
@@ -12,7 +13,7 @@ class FuzzyScorer {
     }
     return when (term distanceTo field) {
       -1 -> NO_MATCH
-      0 -> NO_MATCH
+      0 -> EXACT_MATCH
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/search/FuzzyScorer.kt:16]: change to make 0 difference behave as expected

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/prisonersearch/PrisonerSearchApiClient.kt [hunk 3]
```diff
@@ -24,10 +24,8 @@ class PrisonerSearchApiClient(@Qualifier("prisonerSearchApiWebClient") private v
       ?: emptyList()
   }
 
-  fun getPrisoner(prisonNumber: String): MatchPrisonerResponse? {
-    return matchPrisoner(MatchPrisonerRequest(prisonNumber))
-      ?.firstOrNull()
-  }
+  fun getPrisoner(prisonNumber: String): MatchPrisonerResponse? =
+    matchPrisoner(MatchPrisonerRequest(prisonNumber)).firstOrNull()
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/prisonersearch/PrisonerSearchApiClient.kt:28]: slight tidy up of a warning

File: src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/RecentArrivalsServiceTest.kt [hunk 4]
```diff
@@ -22,16 +23,14 @@ class RecentArrivalsServiceTest {
...
-      0
+      LocalDate.of(2017, 1, 2) to LocalDate.of(2020, 1, 2),
+      PageRequest.of(0, 50)
     )
     val firstElement = arrivals.get().findFirst().get()
     val lastElement = arrivals.get().skip(arrivals.get().count() - 1).findFirst().get()
 
-    assertThat(firstElement.prisonNumber).isEqualTo("AA000")
-    assertThat(lastElement.prisonNumber).isEqualTo("AA0049")
+    assertThat(firstElement.prisonNumber).isEqualTo("AA00000")
```
Comment (andrewrlee) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/RecentArrivalsServiceTest.kt:32]: padded these with 0's to ensure we could sort lexographically

File: src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resource/RecentArrivalsResourceTest.kt [hunk 5]
```diff
@@ -100,6 +100,46 @@ class RecentArrivalsResourceTest : IntegrationTestBase() {
...
+
+  @Test
+  fun `applies search query correctly`() {
+    prisonApiMockServer.stubGetMovementSuccess(
+      "MDI",
+      LocalDate.of(2019, 1, 2).atStartOfDay(),
+      LocalDate.of(2020, 1, 2).atTime(LocalTime.MAX)
+    )
+
+    webTestClient.get().uri("/prisons/MDI/recent-arrivals?fromDate=2019-01-02&toDate=2020-01-02&query=HeRKImer")
```
Comment (andrewrlee) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resource/RecentArrivalsResourceTest.kt:112]: only returns single result now due to query

**Review by louisenorris — APPROVED** [review]
LGTM

**Review by GurnankCheema — APPROVED** [review]
LGTM

**Pull request merged**